operator/cmd/mesh/install.go

```diff
@@ -204,7 +204,7 @@ func Install(rootArgs *RootArgs, iArgs *InstallArgs, logOpts *log.Options, stdOu
 		return fmt.Errorf("failed to install manifests: %v", err)
 	}
 
-	if !exists || rev == "" {
+	if !exists || rev == "" && pilotEnabled {
```

Comment:
This prevents webhook manipulation from being an unexpected side effect for things like simply deploying a gateway with an empty profile.

Reply:
Webhook injection is required for (some) gateways though, so this might cause issues...?

Reply:
IIUC this should get done when an istiod is installed, not when just a gateway is being installed. Note that this isn't done at all when `istioctl manifest generate` is used instead of `istioctl install`, so it really should never matter? I must admit that this default webhook code seems pretty complicated and I'm not sure I understand if it's really working properly in all cases

Reply:
Yeah fair point on generate vs install. My concern was

control cluster: install full
remote cluster: install just gateway

But is the 'remote cluster' case still meeting `pilotEnabled` and it is just not literally setting up the Istiod deployment?

Reply:
Hmm, not sure. In the docs, we are using `istioctl manifest generate` on the remote cluster, not sure why. Maybe workaround for another issue? If I have some spare cycles next week, I'll look into this some more.